griptape/loaders/csv_loader.py

```diff
@@ -13,27 +15,34 @@
 class CsvLoader(BaseLoader):
     embedding_driver: Optional[BaseEmbeddingDriver] = field(default=None, kw_only=True)
     delimiter: str = field(default=",", kw_only=True)
+    encoding: str = field(default="utf-8", kw_only=True)
 
-    def load(self, source: str, *args, **kwargs) -> list[CsvRowArtifact]:
-        return self._load_file(source)
-
-    def load_collection(self, sources: list[str], *args, **kwargs) -> dict[str, list[CsvRowArtifact]]:
-        return utils.execute_futures_dict(
-            {utils.str_to_hash(source): self.futures_executor.submit(self._load_file, source) for source in sources}
-        )
-
-    def _load_file(self, filename: str) -> list[CsvRowArtifact]:
+    def load(self, source: bytes | str, *args, **kwargs) -> ErrorArtifact | list[CsvRowArtifact]:
         artifacts = []
 
-        with open(filename, encoding="utf-8") as csv_file:
-            reader = csv.DictReader(csv_file, delimiter=self.delimiter)
-            chunks = [CsvRowArtifact(row) for row in reader]
+        if isinstance(source, bytes):
+            try:
+                source = source.decode(encoding=self.encoding)
+            except UnicodeDecodeError:
+                return ErrorArtifact(f"Failed to decode bytes to string using encoding: {self.encoding}")
+        elif not isinstance(source, str):
+            return ErrorArtifact(f"Expected source to be bytes or string, got {type(source)}")
 
```

Comment:
I think we can remove this type check, let's leave it up to the type checker.

Reply:
I only made this change thanks to the type checker. For some reason that I don't fully understand, when I accept `bytes` as a parameter, it thinks that source can be `bytes | bytearray | memoryview`. If you don't like the `not isinstance(source, str)` I could change this to be:

```py
if isinstance(source, bytes):
    try:
        source = source.decode(encoding=self.encoding)
    except UnicodeDecodeError:
        return ErrorArtifact(f"Failed to decode bytes to string using encoding: {self.encoding}")
elif isinstance(source, bytearray | memoryview):
    return ErrorArtifact(f"Expected source to be bytes or string, got {type(source)}")
```

Reply:
I just went ahead with my alternative suggestion above. (Note that I also needed to use `Union` instead of the pipe syntax to appease pyright's concerns about python version 3.10)

Reply:
Ok, I should have been using `Union` either because it makes no sense to use generics in an expression evaluated at runtime (duh), though it did work locally which is kind of weird. Anyway, it should be fixed now.